### PR TITLE
Drawer search

### PR DESCRIPTION
**Pull request by nplettenberg**

### CODE REVIEW

File: lib/components/screens/search/search_screen.dart [hunk 1]
```diff
@@ -1,63 +1,24 @@
...
-import 'package:harpy/core/core.dart';
-import 'package:harpy/harpy_widgets/harpy_widgets.dart';
-import 'package:harpy/misc/misc.dart';
-import 'package:provider/provider.dart';
 
 class SearchScreen extends StatelessWidget {
-  const SearchScreen();
+  const SearchScreen({
+    Key? key,
+  }) : super(key: key);
```
Comment (robertodoering) [inline lib/components/screens/search/search_screen.dart:8]:
```suggestion
  const SearchScreen();
```
Let's not add a key param if we don't need it

File: lib/components/screens/search/search_screen.dart [hunk 2]
```diff
@@ -1,63 +1,24 @@
...
-            child: defaultVerticalSpacer,
-          ),
-          const TrendsList(),
-          SliverToBoxAdapter(
-            child: SizedBox(height: mediaQuery.padding.bottom),
-          ),
-        ],
+    return HarpyScaffold(
+      body: BlocProvider<TrendsBloc>(
+        create: (_) => TrendsBloc()..add(const FindTrendsEvent()),
```
Comment (robertodoering) [inline lib/components/screens/search/search_screen.dart:16]: We already build a trends bloc in `HomeProvider` for the home screen, so we shouldn't build another one here.
Instead you can add the `TrendsBloc` as a parameter and build a `BlocProvider.value` here instead

Comment (nplettenberg) [reply lib/components/screens/search/search_screen.dart]: Done

File: lib/components/screens/search/search_screen.dart (hunk 1 above)
Comment (nplettenberg) [reply lib/components/screens/search/search_screen.dart]: Done

**Review by robertodoering — APPROVED** [review]

**Pull request merged**